uxarray/grid/grid.py

```diff
@@ -915,3 +917,55 @@ def to_shapely_polygons(self, correct_antimeridian_polygons=True):
         """
         polygons = _grid_to_polygons(self, correct_antimeridian_polygons)
         return polygons
+
+    def from_vertices(self, method="spherical_voronoi"):
+        """Create a grid and related information from just vertices, using
+        either Spherical Voronoi or Delaunay Triangulation.
+
+        Parameters
+        ----------
+        method : string, optional
+            Method used to construct a grid from only vertices
+        """
+
+        x = self.Mesh2_node_x.data
+        y = self.Mesh2_node_y.data
+        verts = np.column_stack((x, y))
+
+        if not verts:
+            raise ValueError("No vertices provided")
+
+        if method == "spherical_voronoi":
+            if len(verts) < 4:
+                raise ValueError(
+                    "At least 4 vertices needed for Spherical Voronoi")
+
+            # Normalize longitude values to the range [-180, 180] degrees
+            normalized_verts = []
+            for lon, lat in verts:
+                if -180 <= lon <= 180:
+                    normalized_verts.append((lon, lat))
+                else:
+                    normalized_verts.append((lon - 360, lat))
+
+            # Convert the normalized vertices back to a NumPy array
+            normalized_verts = np.array(normalized_verts)
+
+            # Perform Spherical Voronoi Construction
+            grid = SphericalVoronoi(normalized_verts)
+
+            # Handle special cases near the antimeridian and poles if necessary
+            # (e.g., split Voronoi cells that cross the antimeridian)
+
+            # TODO: Assign Mesh2 values to the grid
```

Comment:
[This](https://stackoverflow.com/questions/68324079/how-to-truly-calculate-a-spherical-voronoi-diagram-using-sf) shows some example how to do it

It might be easier to do this with cartesian grid.